control-base/src/main/java/fi/nls/oskari/control/layer/GetDataProviderHandler.java

```diff
@@ -0,0 +1,39 @@
+package fi.nls.oskari.control.layer;
+
+import org.json.JSONObject;
+import org.oskari.service.util.ServiceFactory;
+
+import fi.nls.oskari.annotation.OskariActionRoute;
+import fi.nls.oskari.control.ActionException;
+import fi.nls.oskari.control.ActionHandler;
+import fi.nls.oskari.control.ActionParameters;
+import fi.nls.oskari.domain.map.DataProvider;
+import fi.nls.oskari.map.layer.DataProviderService;
+import fi.nls.oskari.util.JSONHelper;
+import fi.nls.oskari.util.ResponseHelper;
+
+/**
+ * Get single data provider registered in Oskari database
+ */
+@OskariActionRoute("GetDataProvider")
+public class GetDataProviderHandler extends ActionHandler {
+
+	private static final String PARAM_ID = "id";
+
+	private DataProviderService service = ServiceFactory.getDataProviderService();
+
+	@Override
+	public void handleAction(ActionParameters params) throws ActionException {
+		try {
+			final int id = Integer.valueOf(params.getHttpParam(PARAM_ID));
+			DataProvider d = service.find(id);
+			final JSONObject result = new JSONObject();
+			JSONHelper.putValue(result, "dataprovider", d.getAsJSON());
```

Comment:
I dont think the response needs to we wrapped in an additional JSON object. Just return the getAsJSON() as is? Also consider renaming this to DataProvider extending RestActionHandler so we could combine the save/delete to this as well and get rid of https://github.com/oskariorg/oskari-server/blob/master/control-base/src/main/java/fi/nls/oskari/control/layer/SaveOrganizationHandler.java etc

Reply:
Fixed but left mentioned combination of save and delete to future